osu.Game/Utils/ModUtils.cs

```diff
@@ -115,7 +115,9 @@ public static bool CheckValidForGameplay(IEnumerable<Mod> mods, [NotNullWhen(fal
         {
             mods = mods.ToArray();
 
-            CheckCompatibleSet(mods, out invalidMods);
+            // exclude multi mods from compatibility checks.
+            // the loop below automatically marks all multi mods as not valid for gameplay anyway.
+            CheckCompatibleSet(mods.Where(m => !(m is MultiMod)), out invalidMods);
```

Comment:
Without this change, the following test failure surfaces:

```
  Failed TestInvalidModScenarios([osu.Game.Rulesets.Mods.MultiMod, osu.Game.Rulesets.Osu.Mods.OsuModDaycore],[osu.Game.Rulesets.Mods.MultiMod]) [17 ms]
  Error Message:
     Expected: equivalent to < <osu.Game.Rulesets.Mods.MultiMod> >
  But was:  < <osu.Game.Rulesets.Osu.Mods.OsuModDaycore>, <osu.Game.Rulesets.Osu.Mods.OsuModHalfTime>, <osu.Game.Rulesets.Mods.MultiMod> >
  Extra (2): < <osu.Game.Rulesets.Osu.Mods.OsuModDaycore>, <osu.Game.Rulesets.Osu.Mods.OsuModHalfTime> >
```

This is happening because `CheckCompatibleSet()` is flattening the multi mods and deciding that it should report all mods within the multi mod as incompatible, which runs counter to intent of this method..